### ISSUE TITLE
linter/oxc/bad-char-at-comparison: expand coverage

### ISSUE THREAD

**Issue by baevm** [oxc-project/oxc#24990]
Some cases [unicorn/no-invalid-character-comparison](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v72.0.0/docs/rules/no-invalid-character-comparison.md) catches but `oxc/bad-char-at-comparison` misses, I think oxc rule could report them too:

1. `charAt()` without an argument:

```js
"abc".charAt() === "ab";
```

2. `String#at()`:

```js
"abc".at(0) === "ab";
"abc".at(-1) !== "ab";
"abc".at() === "ab";
```

3. String bracket access:

```js
"abc"[0] === "ab";
"abc"["0"] === "ab";

declare const value: string;
value[0] === "ab";
```

4. Template-literal comparands:

```js
"abc".charAt(0) === `ab`;
```

5. Parenthesized access or comparand:

```js
("abc".charAt(0)) === "ab";
"abc".charAt(0) === ("ab");
```

6. Astral Unicode characters such as emoji:

```js
"abc".charAt(0) === "😀";
"abc".charAt(0) === "\u{1F600}";
```

JavaScript evaluates `"😀".length` as 2 because the emoji consists of two UTF-16 code units. Oxc counts Unicode scalar values using Rust’s character semantics and therefore considers it one character.

**Comment by camc314** [comment]
> charAt() without an argument:

lets let TS catch this case.


Agree with all the rest :+1:
